### PR TITLE
Avalonia - Couple fixes and improvements

### PR DESCRIPTION
**Pull request by emmauss**
Includes;

1.Fix crash with the updater on launch.
2.Reduce size of cheat window
3.Enable Tiered Compilation
4.Remove warnings from the avalonia project

### CODE REVIEW

File: Ryujinx.Ava/Ui/Windows/MainWindow.axaml.cs [hunk 1]
```diff
@@ -536,6 +534,14 @@ private void InitializeComponent()
             LoadHotKeys();
         }
 
+        protected override void OnOpened(EventArgs e)
+        {
+            base.OnOpened(e);
+
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/Windows/MainWindow.axaml.cs:540]: Double empty line.

**Review by gdkchan — COMMENTED** [review]
Maybe its worth addressing the remaining issues here too to avoid opening many small PRs. I'm talking about the compiler warnings, and re-enabling tiered compilation for Avalonia.

File: Ryujinx.Ava/Ui/Windows/MainWindow.axaml.cs (hunk 1 above)
Comment (emmauss) [reply Ryujinx.Ava/Ui/Windows/MainWindow.axaml.cs:540]: fixed

File: Ryujinx.Ava/Common/ApplicationHelper.cs [hunk 2]
```diff
@@ -77,8 +78,11 @@ public static void Initialize(VirtualFileSystem virtualFileSystem, AccountManage
 
                 if (result.IsFailure())
                 {
-                    ContentDialogHelper.CreateErrorDialog(_owner,
+                    Dispatcher.UIThread.Post(async () =>
+                    {
+                        await ContentDialogHelper.CreateErrorDialog(_owner,
                                                           string.Format(LocaleManager.Instance["DialogMessageCreateSaveErrorMessage"], result.ToStringWithName()));
```
Comment (gdkchan) [inline Ryujinx.Ava/Common/ApplicationHelper.cs:84]: Indentation is off here.

File: Ryujinx.Ava/Ryujinx.Ava.csproj [hunk 3]
```diff
@@ -5,8 +5,8 @@
     <OutputType>Exe</OutputType>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Version>1.0.0-dirty</Version>
-    <TieredCompilation>false</TieredCompilation>
-    <TieredCompilationQuickJit>false</TieredCompilationQuickJit>
+    <TieredCompilation>true</TieredCompilation>
```
Comment (gdkchan) [inline Ryujinx.Ava/Ryujinx.Ava.csproj:8]: Those two can be just removed, since the default is enabled.

File: Ryujinx.Ava/Ryujinx.Ava.csproj [hunk 4]
```diff
@@ -15,6 +15,7 @@
   <PropertyGroup Condition="'$(RuntimeIdentifier)' != ''">
     <PublishSingleFile>true</PublishSingleFile>
     <PublishTrimmed>true</PublishTrimmed>
+	<PublishReadyToRun>true</PublishReadyToRun>
```
Comment (gdkchan) [inline Ryujinx.Ava/Ryujinx.Ava.csproj:18]:
```suggestion
    <PublishReadyToRun>true</PublishReadyToRun>
```
Tab to spaces.

File: Ryujinx.Ava/Ui/Windows/CheatWindow.axaml [hunk 5]
```diff
@@ -56,9 +62,9 @@
                         </StackPanel>
                     </TreeDataTemplate>
                     <DataTemplate x:DataType="model:CheatModel">
-                        <StackPanel Orientation="Horizontal" HorizontalAlignment="Left">
-                            <CheckBox IsChecked="{Binding IsEnabled}" MinWidth="20" />
-                            <TextBlock Text="{Binding CleanName}" />
+                        <StackPanel Orientation="Horizontal" Margin="0" HorizontalAlignment="Left">
+                            <CheckBox IsChecked="{Binding IsEnabled}" Padding="0" Margin="5, 0" MinWidth="20" />
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/Windows/CheatWindow.axaml:66]:
```suggestion
                            <CheckBox IsChecked="{Binding IsEnabled}" Padding="0" Margin="5,0" MinWidth="20" />
```

Comment (gdkchan) [reply Ryujinx.Ava/Ui/Windows/CheatWindow.axaml:66]: As far I can tell, the others doesn't have this space after `,`.

File: Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <RuntimeIdentifiers>win10-x64;osx-x64;linux-x64</RuntimeIdentifiers>
     <OutputType>Exe</OutputType>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Version>1.0.0-dirty</Version>
-    <TieredCompilation>false</TieredCompilation>
-    <TieredCompilationQuickJit>false</TieredCompilationQuickJit>
+	<TieredCompilation>true</TieredCompilation>
```
Comment (gdkchan) [inline Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj:9]: Same comment as before.

File: Ryujinx/Ryujinx.csproj [hunk 7]
```diff
@@ -16,9 +16,11 @@
   <PropertyGroup Condition="'$(RuntimeIdentifier)' != ''">
     <PublishSingleFile>true</PublishSingleFile>
     <PublishTrimmed>true</PublishTrimmed>
+	<PublishReadyToRun>true</PublishReadyToRun>
   </PropertyGroup>
 
-  <ItemGroup>
+
```
Comment (gdkchan) [inline Ryujinx/Ryujinx.csproj:22]: Double empty line and indentation is off (maybe another case of it using Tabs rather than spaces).

**Comment by gdkchan** [comment]
From what I'm seeing from docs, ReadyToRun significantly increases the binary size. Have you compared them and have some numbers to show for the executable size? And does it provide a significant improvement to justify the size increase?

**Comment by emmauss** [comment]
For ava, the folder is 171MB, and the executable 153MB

**Comment by emmauss** [comment]
Addressed review

**Comment by gdkchan** [comment]
The csproj files were changed from spaces to tab, while it should be the opposite.

**Comment by emmauss** [comment]
> The csproj files were changed from spaces to tab, while it should be the opposite.

It's corrected

**Comment by gdkchan** [comment]
> > The csproj files were changed from spaces to tab, while it should be the opposite.
> 
> It's corrected

The axaml files seems to have the same issue.

**Comment by emmauss** [comment]
> > > The csproj files were changed from spaces to tab, while it should be the opposite.
> > 
> > 
> > It's corrected
> 
> The axaml files seems to have the same issue.

That's also fixed

File: Ryujinx/Ryujinx.csproj (hunk 7 above)
Comment (gdkchan) [reply Ryujinx/Ryujinx.csproj:20]: This was not fixed.

Comment (emmauss) [reply Ryujinx/Ryujinx.csproj]: I should be fixed now.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Comment by gdkchan** [comment]
One thing to note is that this is also enabling tiered compilation on the other projects (headless and GTK UI).. So it could also have a small improvement on startup time, but I doubt it would be as noticeable as Avalonia.

**Review by AcK77 — APPROVED** [review]

**Pull request merged**